front/app/modules/commercial/insights/hooks/useInsightsCategorySuggestions.test.tsx

```diff
@@ -0,0 +1,233 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import useInsightsCategorySuggestions, {
+  QueryParameters,
+} from './useInsightsCategorySuggestions';
+import { Observable, Subscription } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import { waitFor } from 'utils/testUtils/rtl';
+import { insightsCategorySuggestionsStream } from 'modules/commercial/insights/services/insightsCategorySuggestions';
+import streams from 'utils/streams';
+
+const viewId = '1';
+
+const queryParameters: QueryParameters = {
+  categories: ['3'],
+  inputs: ['10'],
+};
+
+const mockCategorySuggestions = {
+  data: [
+    {
+      id: '58ed4a03-155b-4b60-ac9e-cf101e6d94d0',
+      type: 'zeroshot_classification_task',
+      attributes: {
+        created_at: '2021-07-08T12:01:53.254Z',
+      },
+      relationships: {
+        categories: {
+          data: [
+            {
+              id: 'e499e92b-3d9a-4147-99ed-4abdc4fe558f',
+              type: 'category',
+            },
+          ],
+        },
+        inputs: {
+          data: [
+            {
+              id: '20c7e056-7c7c-477f-bf0e-72a7ce6fb515',
+              type: 'input',
+            },
+          ],
+        },
+      },
+    },
+    {
+      id: '140b1468-8b49-4999-a51c-084d8e17eefa',
+      type: 'zeroshot_classification_task',
+      attributes: {
+        created_at: '2021-07-08T12:01:53.330Z',
+      },
+      relationships: {
+        categories: {
+          data: [
+            {
+              id: 'e499e92b-3d9a-4147-99ed-4abdc4fe558f',
+              type: 'category',
+            },
+          ],
+        },
+        inputs: {
+          data: [
+            {
+              id: 'e8b3aa62-4ea2-474b-a97b-872e7ca47b73',
+              type: 'input',
+            },
+          ],
+        },
+      },
+    },
+  ],
+};
+
+let mockObservable = new Observable((subscriber) => {
+  subscriber.next(mockCategorySuggestions);
+}).pipe(delay(1));
+
+jest.mock(
+  'modules/commercial/insights/services/insightsCategorySuggestions',
+  () => {
+    return {
+      insightsCategorySuggestionsStream: jest.fn(() => {
+        return {
+          observable: mockObservable,
+        };
+      }),
+    };
+  }
+);
+
+jest.mock('utils/streams', () => {
+  return { fetchAllWith: jest.fn() };
+});
+
+jest.mock('utils/helperUtils', () => {
+  return { isNilOrError: jest.fn(() => false) };
+});
+
+describe('useInsightsCategorySuggestions', () => {
+  it('should call useInsightsCategorySuggestions with correct viewId', async () => {
+    renderHook(() => useInsightsCategorySuggestions(viewId));
+    expect(insightsCategorySuggestionsStream).toHaveBeenCalledWith(viewId, {
```

Comment:
I would like to question the naming of this stream and the affiliated hook.
- Neither returns what the name would suggest : I would expect category suggestions (objects that I don't think we expose for now, but I'd imagine like something with an input and a category), or inputs with linked categories as category suggestions, instead you get tasks. 
- You have no way to guess from the name of the hook that some polling's going on and refreshed the inputs with new categories suggested.

Since polling's rare, I think we should make the name reflect it for discoverability. 
Since tasks could be used in themselves, they're a progress indicator, I think we should rename the stream.
I'm bad at naming, so I'll just give it a try : insightsCategorySuggestionsTasksStream (a bit of a mouthful) and triggerSuggestCategoriesPollInputs (even worse readable) or suggestCategoriesForInputs (kind of vague and looses the tasks) or useInsightsCatgeoriesSuggestionsTasks (clear, and too bad for the polling ?)
The task is arduous, but I think we can name them better

Reply:
Agree on the naming issue, it's not the best right now... From your suggestions, I think I like `useInsightsCatgeoriesSuggestionsTasks` the most as it is consistent with how we name the rest of the things. The polling is more of an implementation detail, I personally don't think it should be reflected in the name.